### PR TITLE
Allow retries for pull

### PR DESCRIPTION
**Pull request by bkelley17**
Currently the plugin allows configurable retries for push, but not for pull. This change allows retries to be configured for pull as well.

### CODE REVIEW

**Comment by bkelley17** [comment]
I wonder if someone can help me understand what the codecov/patch check is that failed. Is that because coverage of introduced code is not high enough?

**Comment by rhuss** [comment]
@bkelley17 yes, it's because that we generally require that new code should be at least as much tested then the existing code. Not really sure how this is measure as I would expect than if the overall coverage gets increased, then the contribution must be at least have a higher coverage than the rest.

But don't worry too much about it, we also do exceptions from the rule.
Please give me some time to do a review, I hope I can jump on it this week.

thanks for your contribution ...

File: src/main/java/io/fabric8/maven/docker/BuildMojo.java [hunk 1]
```diff
@@ -26,6 +26,9 @@
     @Parameter(property = "docker.skip.build", defaultValue = "false")
     protected boolean skipBuild;
 
+    @Parameter(property = "docker.pull.retries", defaultValue = "0")
+    private int retries;
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/BuildMojo.java]: Could we please rename this in `pullRetries` for the member configuration ? "retries" alone is a bit unspecific.

Comment (rhuss) [reply src/main/java/io/fabric8/maven/docker/BuildMojo.java]: Same for the other MoJos please and the documentation.

**Comment by rhuss** [comment]
I will work on this PR for the next release (resolving the conflicts) and add the suggestions as I think this is a valuable feature.

**Comment by rohanKanojia** [comment]
Superseded by #1697

**Pull request closed without merging**